src/3rd_party.c

```diff
@@ -0,0 +1,173 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2019 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#define _GNU_SOURCE
+#include "3rd_party.h"
+#include "config.h"
+#include "lib/log.h"
+#include "swupd.h"
+
+static char *cmdline_option_repo = NULL;
+
+#define CHOOSE_ADD 9000
+#define CHOOSE_LIST 9001
+#define CHOOSE_REMOVE 9003
+static bool cmdline_option_add;
+static bool cmdline_option_list;
+static bool cmdline_option_remove;
+
+#ifdef THIRDPARTY
+
+void get_repo_path(char **repo_path, char *repo_name)
+{
+	string_or_die(repo_path, "%s%s", THIRDPARTY_REPO_PREFIX, repo_name);
+}
+
+static void print_help(void)
+{
+	print("Usage:\n");
+	print("   swupd 3rd-party [OPTION...]\n\n");
+
+	global_print_help();
+
+	print("Options:\n");
+	print("   --add <repo_name>		Add third party repo\n");
```

Comment:
I don't think it's a good idea to use flags for add, list remove. If you use a flag we need to protect against multiple flags used together and this will become hard after adding bundle-add and other commands

Reply:
> I don't think it's a good idea to use flags for add, list remove. If you use a flag we need to protect against multiple flags used together and this will become hard after adding bundle-add and other commands

I was thinking about that....I need to have add, list, remove as sub_commands like add in git remote add

Reply:
Makes sense will change in next push